### DISCUSSION TITLE
Define boundary conditions in Hinged Euler–Bernoulli Beam

### DISCUSSION THREAD

**Discussion by LiutianZhang** [zwicker-group/py-pde#608] [Q&A]
Thank you for your helpful work! I tried to be familiar with the package by a simple Euler–Bernoulli Beam. Hinged boundary conditions are used at both ends, which means $y(0,t)=0$, $y(L,t)=0$, $\frac{\partial^2y(0,t)}{\partial x^2}=0$ and $\frac{\partial^2y(L,t)}{\partial x^2}=0$. Let $x_1 = y$ and $x_2=\frac{\partial y}{\partial t}$, boundary conditions are defined by 'bc_ops', but it shows that 'Unused BCs: ['x1:*']', how can I implement such boundary conditions in this package?

$$m \frac{\partial^2 y(x,t)}{\partial t^2}+ c \frac{\partial y(x,t)}{\partial t}+ EI\frac{\partial^2}{\partial x^2}\left(\frac{\partial^2y(x,t)}{\partial x^2}\right)=0$$

```
from pde import PDE, FieldCollection, ScalarField, CartesianGrid
import math

m = 1
c = 0.05
EI = 1
eq = PDE(
    {"x1": f"x2",
      "x2": f"-{c/m}*x2-{EI/m}*laplace(laplace(x1))"
    },
    bc_ops=
    {
        'x1:*': {"curvature": 0},
        'x1:*': {"value": 0}
    }
)


# initialize state
xnum = 50
L = 1
grid = CartesianGrid([[0,L]],xnum)

statex1 = ScalarField.from_expression(grid, f"0.1*sin({math.pi / L}*x)")
statex2 = ScalarField.from_expression(grid, f"0")

state = FieldCollection([statex1, statex2])

# simulate the pde
t_interval = 1e-3
trackers = [
"progress",
# show progress bar during simulation
]
tmax = 1
sol = eq.solve(state, t_range=tmax, dt=1e-5, tracker = trackers, solver = 'implicit')
```

**Comment by david-zwicker** [comment]
Well, the current approach cannot work since you use the same key twice in the dictionary for the `bc_ops` argument. Are you trying to provide different BCs for the two different laplace operators in the definition of the rate equation for `x2`?

**Reply by LiutianZhang** [reply]
Thank you. I tried to define different BCs for x1 and x2, but it still show that 'Unused BCs: ['x1:*']', i don't know when a defined BC will be ignored in the package.
```
    bc_ops=
    {
        'x1:*': {"curvature": 0},
        'x2:*': {"value": 0}
    }
```

**Reply by david-zwicker** [reply]
The BC for `'x1:*'` defines boundary conditions in the right-hand side of the evolution rate of `x1`. However, since there are no differential operators in this right hand side, the boundary condition cannot used. The package warns you here to prevent mistakes.

**Reply by LiutianZhang** [reply]
Thank you for your answer!
